### PR TITLE
Remove HdfsUnderFileSystemFactory from service loading in COSN UFS jar

### PR DESCRIPTION
**Pull request by Haoning-Sun**
### What changes are proposed in this pull request?

Remove HdfsUnderFileSystemFactory from service loading in COSN UFS jar.

### Why are the changes needed?
Avoid loading HdfsUnderFileSystemFactory from alluxio-underfs-cosn-*.jar.

### Does this PR introduce any user facing changes?
No.

### CODE REVIEW

**Review by elega — APPROVED** [review]

**Comment by elega** [comment]
@jiacheliu3 can you help merge?

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

File: underfs/cosn/pom.xml [hunk 1]
```diff
@@ -69,6 +69,37 @@
...
+                  <filter>
+                    <artifact>*:*</artifact>
+                    <excludes>
+                      <exclude>LICENSE</exclude>
+                      <exclude>META-INF/LICENSE</exclude>
+                      <exclude>META-INF/*.SF</exclude>
+                      <exclude>META-INF/*.DSA</exclude>
+                      <exclude>META-INF/*.RSA</exclude>
+                      <!-- Because this module depends on alluxio-underfs-hdfs, eliminate HDFS factory implementation -->
+                      <exclude>alluxio/underfs/hdfs/HdfsUnderFileSystemFactory.*</exclude>
```
Comment (maobaolong) [inline underfs/cosn/pom.xml:90]: If you exclude this, the cosn module would encountered a ClassNotFoundException.

What you want to do is exclude `META-INF/services/alluxio.underfs.UnderFileSystemFactory` only.

Comment (maobaolong) [reply underfs/cosn/pom.xml:90]: /ping @elega @jiacheliu3

**Pull request merged**